src/renderers/dom/client/wrappers/ReactDOMSelect.js

```diff
@@ -150,6 +161,8 @@ var ReactDOMSelect = {
       checkSelectPropTypes(inst, props);
     }
 
+    warnIfValueIsNull(props);
```

Comment:
This should go into the `if(__DEV__)` check above.  We only fire warnings in dev mode.